// @react-spectrum/color/dist/ColorSwatchPicker.main.js
require("./ColorSwatchPicker.1575be06.css");
var $f076f5c85640192a$exports = require("./ColorSwatch.main.js");
var $iY2T3$reactariacomponents = require("react-aria-components");
var $iY2T3$react = require("react");
var $iY2T3$reactspectrumutils = require("@react-spectrum/utils");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ColorSwatchPicker", () => $ba9a184bcae9b225$export$b46792416e3d8515);
/*
 * Copyright 2024 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



function $ba9a184bcae9b225$var$ColorSwatchPicker(props, ref) {
    let { density: density = 'regular', size: size = 'M', rounding: rounding = 'none', ...otherProps } = props;
    let { styleProps: styleProps } = (0, $iY2T3$reactspectrumutils.useStyleProps)(props);
    let domRef = (0, $iY2T3$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($iY2T3$react))).createElement((0, $iY2T3$reactariacomponents.ColorSwatchPicker), {
        ...otherProps,
        style: styleProps.style,
        ref: domRef,
        className: styleProps.className + function anonymous(props) {
            let rules = "";
            rules += ' _Sd';
            rules += ' _0a';
            if (props.density === "spacious") rules += ' lc';
            else if (props.density === "regular") rules += ' lb';
            else if (props.density === "compact") rules += ' lF';
            if (props.density === "spacious") rules += ' mc';
            else if (props.density === "regular") rules += ' mb';
            else if (props.density === "compact") rules += ' mF';
            return rules;
        }({
            density: density
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($iY2T3$react))).createElement((0, $f076f5c85640192a$exports.SpectrumColorSwatchContext).Provider, {
        value: {
            useWrapper: $ba9a184bcae9b225$var$useWrapper,
            size: size,
            rounding: rounding
        }
    }, props.children));
}
/**
 * A ColorSwatchPicker displays a list of color swatches and allows a user to select one of them.
 */ let $ba9a184bcae9b225$export$b46792416e3d8515 = /*#__PURE__*/ (0, $iY2T3$react.forwardRef)($ba9a184bcae9b225$var$ColorSwatchPicker);
function $ba9a184bcae9b225$var$useWrapper(swatch, color, rounding) {
    return /*#__PURE__*/ (0, ($parcel$interopDefault($iY2T3$react))).createElement((0, $iY2T3$reactariacomponents.ColorSwatchPickerItem), {
        color: color,
        className: (renderProps)=>(function anonymous(props) {
                let rules = "";
                if (props.isFocusVisible) rules += ' _Eb';
                else rules += ' _Ea';
                rules += ' da___D';
                rules += ' d___I';
                rules += ' _Gc';
                rules += ' _Fc';
                rules += ' Xc';
                if (props.rounding === "full") rules += ' _pf';
                else if (props.rounding === "default") rules += ' _pc';
                else if (props.rounding === "none") rules += ' _pa';
                if (props.rounding === "full") rules += ' _qf';
                else if (props.rounding === "default") rules += ' _qc';
                else if (props.rounding === "none") rules += ' _qa';
                if (props.rounding === "full") rules += ' _rf';
                else if (props.rounding === "default") rules += ' _rc';
                else if (props.rounding === "none") rules += ' _ra';
                if (props.rounding === "full") rules += ' _sf';
                else if (props.rounding === "default") rules += ' _sc';
                else if (props.rounding === "none") rules += ' _sa';
                return rules;
            })({
                ...renderProps,
                rounding: rounding
            })
    }, ({ isSelected: isSelected })=>/*#__PURE__*/ (0, ($parcel$interopDefault($iY2T3$react))).createElement((0, ($parcel$interopDefault($iY2T3$react))).Fragment, null, swatch, isSelected && /*#__PURE__*/ (0, ($parcel$interopDefault($iY2T3$react))).createElement("div", {
            "aria-hidden": true,
            className: function anonymous(props) {
                let rules = "";
                rules += ' Xa';
                rules += ' __ya';
                rules += ' _0a';
                rules += ' _2a';
                rules += ' _1a';
                rules += ' _3a';
                rules += ' _o-1icpv9v';
                rules += ' _tb';
                rules += ' _p-17zqamw';
                rules += ' _q-17zqamw';
                rules += ' _r-17zqamw';
                rules += ' _s-17zqamw';
                return rules;
            }()
        })));
}
