// @react-spectrum/color/dist/ColorSwatch.module.js
import "./ColorSwatch.f6e6f811.css";
import {useColorSwatch as $5Pjc1$useColorSwatch} from "@react-aria/color";
import {useContextProps as $5Pjc1$useContextProps, ColorSwatchContext as $5Pjc1$ColorSwatchContext} from "react-aria-components";
import $5Pjc1$react, {createContext as $5Pjc1$createContext, useContext as $5Pjc1$useContext, forwardRef as $5Pjc1$forwardRef} from "react";
import {useDOMRef as $5Pjc1$useDOMRef, useStyleProps as $5Pjc1$useStyleProps} from "@react-spectrum/utils";

/*
 * Copyright 2024 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



const $d68786ae409fa722$export$8529d7908a78c058 = /*#__PURE__*/ (0, $5Pjc1$createContext)(null);
function $d68786ae409fa722$var$ColorSwatch(props, ref) {
    let domRef = (0, $5Pjc1$useDOMRef)(ref);
    [props, domRef] = (0, $5Pjc1$useContextProps)(props, domRef, (0, $5Pjc1$ColorSwatchContext));
    let { colorSwatchProps: colorSwatchProps, color: color } = (0, $5Pjc1$useColorSwatch)(props);
    let { styleProps: styleProps } = (0, $5Pjc1$useStyleProps)(props);
    let ctx = (0, $5Pjc1$useContext)($d68786ae409fa722$export$8529d7908a78c058);
    let { size: size = (ctx === null || ctx === void 0 ? void 0 : ctx.size) || 'M', rounding: rounding = (ctx === null || ctx === void 0 ? void 0 : ctx.rounding) || 'default' } = props;
    let swatch = /*#__PURE__*/ (0, $5Pjc1$react).createElement("div", {
        ...colorSwatchProps,
        ...styleProps,
        ref: domRef,
        style: {
            ...styleProps.style,
            // TODO: should there be a distinction between transparent and no value (e.g. null)?
            background: color.getChannelValue('alpha') > 0 ? `linear-gradient(${color}, ${color}), repeating-conic-gradient(#e6e6e6 0% 25%, white 0% 50%) 0% 50% / 16px 16px` : 'linear-gradient(to bottom right, transparent calc(50% - 2px), var(--spectrum-red-900) calc(50% - 2px) calc(50% + 2px), transparent calc(50% + 2px)) no-repeat'
        },
        className: styleProps.className + function anonymous(props) {
            let rules = "";
            if (props.size === "L") rules += ' ok';
            else if (props.size === "M") rules += ' oi';
            else if (props.size === "S") rules += ' og';
            else if (props.size === "XS") rules += ' oe';
            if (props.size === "L") rules += ' nk';
            else if (props.size === "M") rules += ' ni';
            else if (props.size === "S") rules += ' ng';
            else if (props.size === "XS") rules += ' ne';
            if (props.rounding === "full") rules += ' _pf';
            else if (props.rounding === "none") rules += ' _pa';
            else if (props.rounding === "default") rules += ' _pc';
            if (props.rounding === "full") rules += ' _qf';
            else if (props.rounding === "none") rules += ' _qa';
            else if (props.rounding === "default") rules += ' _qc';
            if (props.rounding === "full") rules += ' _rf';
            else if (props.rounding === "none") rules += ' _ra';
            else if (props.rounding === "default") rules += ' _rc';
            if (props.rounding === "full") rules += ' _sf';
            else if (props.rounding === "none") rules += ' _sa';
            else if (props.rounding === "default") rules += ' _sc';
            rules += ' c-1n5whoe';
            rules += ' wb';
            rules += ' xb';
            rules += ' ub';
            rules += ' vb';
            rules += ' Aa';
            rules += ' __ka';
            rules += ' _tb';
            return rules;
        }({
            size: size,
            rounding: rounding
        })
    });
    // ColorSwatchPicker needs to wrap the swatch in a ListBoxItem.
    if (ctx) return ctx.useWrapper(swatch, color, rounding);
    return swatch;
}
/**
 * A ColorSwatch displays a preview of a selected color.
 */ let $d68786ae409fa722$export$cae13e90592f246a = /*#__PURE__*/ (0, $5Pjc1$forwardRef)($d68786ae409fa722$var$ColorSwatch);


export {$d68786ae409fa722$export$8529d7908a78c058 as SpectrumColorSwatchContext, $d68786ae409fa722$export$cae13e90592f246a as ColorSwatch};
