// @react-spectrum/color/dist/ColorSwatch.main.js
require("./ColorSwatch.f6e6f811.css");
var $ikjCA$reactariacolor = require("@react-aria/color");
var $ikjCA$reactariacomponents = require("react-aria-components");
var $ikjCA$react = require("react");
var $ikjCA$reactspectrumutils = require("@react-spectrum/utils");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "SpectrumColorSwatchContext", () => $f076f5c85640192a$export$8529d7908a78c058);
$parcel$export(module.exports, "ColorSwatch", () => $f076f5c85640192a$export$cae13e90592f246a);
/*
 * Copyright 2024 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



const $f076f5c85640192a$export$8529d7908a78c058 = /*#__PURE__*/ (0, $ikjCA$react.createContext)(null);
function $f076f5c85640192a$var$ColorSwatch(props, ref) {
    let domRef = (0, $ikjCA$reactspectrumutils.useDOMRef)(ref);
    [props, domRef] = (0, $ikjCA$reactariacomponents.useContextProps)(props, domRef, (0, $ikjCA$reactariacomponents.ColorSwatchContext));
    let { colorSwatchProps: colorSwatchProps, color: color } = (0, $ikjCA$reactariacolor.useColorSwatch)(props);
    let { styleProps: styleProps } = (0, $ikjCA$reactspectrumutils.useStyleProps)(props);
    let ctx = (0, $ikjCA$react.useContext)($f076f5c85640192a$export$8529d7908a78c058);
    let { size: size = (ctx === null || ctx === void 0 ? void 0 : ctx.size) || 'M', rounding: rounding = (ctx === null || ctx === void 0 ? void 0 : ctx.rounding) || 'default' } = props;
    let swatch = /*#__PURE__*/ (0, ($parcel$interopDefault($ikjCA$react))).createElement("div", {
        ...colorSwatchProps,
        ...styleProps,
        ref: domRef,
        style: {
            ...styleProps.style,
            // TODO: should there be a distinction between transparent and no value (e.g. null)?
            background: color.getChannelValue('alpha') > 0 ? `linear-gradient(${color}, ${color}), repeating-conic-gradient(#e6e6e6 0% 25%, white 0% 50%) 0% 50% / 16px 16px` : 'linear-gradient(to bottom right, transparent calc(50% - 2px), var(--spectrum-red-900) calc(50% - 2px) calc(50% + 2px), transparent calc(50% + 2px)) no-repeat'
        },
        className: styleProps.className + function anonymous(props) {
            let rules = "";
            if (props.size === "L") rules += ' ok';
            else if (props.size === "M") rules += ' oi';
            else if (props.size === "S") rules += ' og';
            else if (props.size === "XS") rules += ' oe';
            if (props.size === "L") rules += ' nk';
            else if (props.size === "M") rules += ' ni';
            else if (props.size === "S") rules += ' ng';
            else if (props.size === "XS") rules += ' ne';
            if (props.rounding === "full") rules += ' _pf';
            else if (props.rounding === "none") rules += ' _pa';
            else if (props.rounding === "default") rules += ' _pc';
            if (props.rounding === "full") rules += ' _qf';
            else if (props.rounding === "none") rules += ' _qa';
            else if (props.rounding === "default") rules += ' _qc';
            if (props.rounding === "full") rules += ' _rf';
            else if (props.rounding === "none") rules += ' _ra';
            else if (props.rounding === "default") rules += ' _rc';
            if (props.rounding === "full") rules += ' _sf';
            else if (props.rounding === "none") rules += ' _sa';
            else if (props.rounding === "default") rules += ' _sc';
            rules += ' c-1n5whoe';
            rules += ' wb';
            rules += ' xb';
            rules += ' ub';
            rules += ' vb';
            rules += ' Aa';
            rules += ' __ka';
            rules += ' _tb';
            return rules;
        }({
            size: size,
            rounding: rounding
        })
    });
    // ColorSwatchPicker needs to wrap the swatch in a ListBoxItem.
    if (ctx) return ctx.useWrapper(swatch, color, rounding);
    return swatch;
}
/**
 * A ColorSwatch displays a preview of a selected color.
 */ let $f076f5c85640192a$export$cae13e90592f246a = /*#__PURE__*/ (0, $ikjCA$react.forwardRef)($f076f5c85640192a$var$ColorSwatch);
