// @react-spectrum/color/dist/ColorSwatchPicker.mjs
import "./ColorSwatchPicker.1575be06.css";
import {SpectrumColorSwatchContext as $d68786ae409fa722$export$8529d7908a78c058} from "./ColorSwatch.mjs";
import {ColorSwatchPicker as $1TpDU$ColorSwatchPicker, ColorSwatchPickerItem as $1TpDU$ColorSwatchPickerItem} from "react-aria-components";
import $1TpDU$react, {forwardRef as $1TpDU$forwardRef} from "react";
import {useStyleProps as $1TpDU$useStyleProps, useDOMRef as $1TpDU$useDOMRef} from "@react-spectrum/utils";

/*
 * Copyright 2024 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



function $4919bc07654f325c$var$ColorSwatchPicker(props, ref) {
    let { density: density = 'regular', size: size = 'M', rounding: rounding = 'none', ...otherProps } = props;
    let { styleProps: styleProps } = (0, $1TpDU$useStyleProps)(props);
    let domRef = (0, $1TpDU$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $1TpDU$react).createElement((0, $1TpDU$ColorSwatchPicker), {
        ...otherProps,
        style: styleProps.style,
        ref: domRef,
        className: styleProps.className + function anonymous(props) {
            let rules = "";
            rules += ' _Sd';
            rules += ' _0a';
            if (props.density === "spacious") rules += ' lc';
            else if (props.density === "regular") rules += ' lb';
            else if (props.density === "compact") rules += ' lF';
            if (props.density === "spacious") rules += ' mc';
            else if (props.density === "regular") rules += ' mb';
            else if (props.density === "compact") rules += ' mF';
            return rules;
        }({
            density: density
        })
    }, /*#__PURE__*/ (0, $1TpDU$react).createElement((0, $d68786ae409fa722$export$8529d7908a78c058).Provider, {
        value: {
            useWrapper: $4919bc07654f325c$var$useWrapper,
            size: size,
            rounding: rounding
        }
    }, props.children));
}
/**
 * A ColorSwatchPicker displays a list of color swatches and allows a user to select one of them.
 */ let $4919bc07654f325c$export$b46792416e3d8515 = /*#__PURE__*/ (0, $1TpDU$forwardRef)($4919bc07654f325c$var$ColorSwatchPicker);
function $4919bc07654f325c$var$useWrapper(swatch, color, rounding) {
    return /*#__PURE__*/ (0, $1TpDU$react).createElement((0, $1TpDU$ColorSwatchPickerItem), {
        color: color,
        className: (renderProps)=>(function anonymous(props) {
                let rules = "";
                if (props.isFocusVisible) rules += ' _Eb';
                else rules += ' _Ea';
                rules += ' da___D';
                rules += ' d___I';
                rules += ' _Gc';
                rules += ' _Fc';
                rules += ' Xc';
                if (props.rounding === "full") rules += ' _pf';
                else if (props.rounding === "default") rules += ' _pc';
                else if (props.rounding === "none") rules += ' _pa';
                if (props.rounding === "full") rules += ' _qf';
                else if (props.rounding === "default") rules += ' _qc';
                else if (props.rounding === "none") rules += ' _qa';
                if (props.rounding === "full") rules += ' _rf';
                else if (props.rounding === "default") rules += ' _rc';
                else if (props.rounding === "none") rules += ' _ra';
                if (props.rounding === "full") rules += ' _sf';
                else if (props.rounding === "default") rules += ' _sc';
                else if (props.rounding === "none") rules += ' _sa';
                return rules;
            })({
                ...renderProps,
                rounding: rounding
            })
    }, ({ isSelected: isSelected })=>/*#__PURE__*/ (0, $1TpDU$react).createElement((0, $1TpDU$react).Fragment, null, swatch, isSelected && /*#__PURE__*/ (0, $1TpDU$react).createElement("div", {
            "aria-hidden": true,
            className: function anonymous(props) {
                let rules = "";
                rules += ' Xa';
                rules += ' __ya';
                rules += ' _0a';
                rules += ' _2a';
                rules += ' _1a';
                rules += ' _3a';
                rules += ' _o-1icpv9v';
                rules += ' _tb';
                rules += ' _p-17zqamw';
                rules += ' _q-17zqamw';
                rules += ' _r-17zqamw';
                rules += ' _s-17zqamw';
                return rules;
            }()
        })));
}


export {$4919bc07654f325c$export$b46792416e3d8515 as ColorSwatchPicker};
