// @react-spectrum/color/dist/ColorPicker.main.js
require("./ColorPicker.ab9f47c0.css");
var $f076f5c85640192a$exports = require("./ColorSwatch.main.js");
var $64NP6$reactariacomponents = require("react-aria-components");
var $64NP6$reactspectrumview = require("@react-spectrum/view");
var $64NP6$reactspectrumdialog = require("@react-spectrum/dialog");
var $64NP6$react = require("react");
var $64NP6$reactspectrumutils = require("@react-spectrum/utils");
var $64NP6$reactariautils = require("@react-aria/utils");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "ColorPicker", () => $0c330b4a4fe337a6$export$9feb1bc2e5f1ccb3);
/*
 * Copyright 2024 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






function $0c330b4a4fe337a6$var$ColorPicker(props, ref) {
    let swatchRef = (0, $64NP6$react.useRef)(null);
    let domRef = (0, $64NP6$reactspectrumutils.useFocusableRef)(ref);
    let labelId = (0, $64NP6$reactariautils.useId)();
    return /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement((0, $64NP6$reactariacomponents.ColorPicker), props, /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement((0, $64NP6$reactspectrumdialog.DialogTrigger), {
        type: "popover",
        mobileType: "tray",
        targetRef: (0, $64NP6$reactspectrumutils.unwrapDOMRef)(swatchRef)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement((0, $64NP6$reactariacomponents.Button), {
        ref: domRef,
        className: function anonymous(props) {
            let rules = "";
            rules += ' ba';
            rules += ' Af';
            rules += ' Ia';
            rules += ' Ja';
            rules += ' Ga';
            rules += ' Ha';
            rules += ' _Sd';
            rules += ' _Uc';
            rules += ' lM';
            rules += ' mM';
            rules += ' _Ea';
            rules += ' _5-bc1l9oh';
            rules += ' _5-1uotwbwg';
            rules += ' _5-eo0c6sf';
            rules += ' _5-enzzrge';
            rules += ' _5-enzykdd';
            rules += ' _5-enzwzjc';
            rules += ' _5-enzrfpb';
            rules += ' _5a';
            rules += ' a___K';
            if (props.size === "L") rules += ' _6d';
            else if (props.size === "M") rules += ' _6c';
            else if (props.size === "S") rules += ' _6b';
            else if (props.size === "XS") rules += ' _6a';
            return rules;
        }({
            size: props.size || 'M'
        })
    }, ({ isFocusVisible: isFocusVisible })=>/*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement((0, ($parcel$interopDefault($64NP6$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement("div", {
            className: function anonymous(props) {
                let rules = "";
                if (props.isFocusVisible) rules += ' _Eb';
                else rules += ' _Ea';
                rules += ' da___D';
                rules += ' d___I';
                rules += ' _Gc';
                rules += ' _Fc';
                rules += ' _pc';
                rules += ' _qc';
                rules += ' _rc';
                rules += ' _sc';
                return rules;
            }({
                isFocusVisible: isFocusVisible
            })
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement((0, $f076f5c85640192a$exports.ColorSwatch), {
            ref: swatchRef,
            color: props.value,
            size: props.size,
            rounding: props.rounding,
            "aria-label": props['aria-label'],
            "aria-labelledby": props['aria-labelledby'],
            "aria-describedby": props['aria-describedby'],
            "aria-details": props['aria-details']
        })), props.label && /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement("span", {
            id: labelId
        }, props.label))), /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement((0, $64NP6$reactspectrumdialog.Dialog), {
        "aria-labelledby": props.label ? labelId : props['aria-labelledby'],
        "aria-label": props['aria-label'],
        UNSAFE_style: {
            width: 'fit-content',
            minWidth: 0,
            margin: '0 auto' // Center within tray.
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).createElement((0, $64NP6$reactspectrumview.Content), {
        UNSAFE_style: {
            position: 'relative',
            margin: 'calc(var(--spectrum-dialog-padding) * -1)',
            padding: 'var(--spectrum-global-dimension-size-200)'
        }
    }, props.children))));
}
/**
 * A ColorPicker combines a swatch with a customizable popover for editing a color.
 */ let $0c330b4a4fe337a6$export$9feb1bc2e5f1ccb3 = /*#__PURE__*/ (0, ($parcel$interopDefault($64NP6$react))).forwardRef($0c330b4a4fe337a6$var$ColorPicker);
