// @react-spectrum/theme-dark/dist/import.mjs
import "./spectrum-dark.e5783241.css";
import $54lWP$vars_spectrumdark_cssmodulejs from "./vars_spectrum-dark_css.mjs";
import "./spectrum-darkest.9999f1da.css";
import $54lWP$vars_spectrumdarkest_cssmodulejs from "./vars_spectrum-darkest_css.mjs";
import "./spectrum-global.674dd93d.css";
import $54lWP$vars_spectrumglobal_cssmodulejs from "./vars_spectrum-global_css.mjs";
import "./spectrum-large.1af55ed7.css";
import $54lWP$vars_spectrumlarge_cssmodulejs from "./vars_spectrum-large_css.mjs";
import "./spectrum-medium.d2b4659b.css";
import $54lWP$vars_spectrummedium_cssmodulejs from "./vars_spectrum-medium_css.mjs";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />





let $88bc786db24c8e69$export$bca14c5b3b88a9c9 = {
    global: ($parcel$interopDefault($54lWP$vars_spectrumglobal_cssmodulejs)),
    light: (0, ($parcel$interopDefault($54lWP$vars_spectrumdark_cssmodulejs))),
    dark: (0, ($parcel$interopDefault($54lWP$vars_spectrumdarkest_cssmodulejs))),
    medium: ($parcel$interopDefault($54lWP$vars_spectrummedium_cssmodulejs)),
    large: ($parcel$interopDefault($54lWP$vars_spectrumlarge_cssmodulejs))
};


export {$88bc786db24c8e69$export$bca14c5b3b88a9c9 as theme};
