// @react-spectrum/theme-light/dist/main.js
require("./spectrum-darkest.761f39dc.css");
var $5fc600550714d659$exports = require("./vars_spectrum-darkest_css.main.js");
require("./spectrum-global.9fc04085.css");
var $33116a1ac1f127f3$exports = require("./vars_spectrum-global_css.main.js");
require("./spectrum-large.80aac790.css");
var $ed6c61a7b1f80a21$exports = require("./vars_spectrum-large_css.main.js");
require("./spectrum-lightest.19121859.css");
var $2179ca4439c0728e$exports = require("./vars_spectrum-lightest_css.main.js");
require("./spectrum-medium.bf2336ac.css");
var $7ada9fcc793cf5cf$exports = require("./vars_spectrum-medium_css.main.js");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "theme", () => $e9051af08b8b1500$export$bca14c5b3b88a9c9);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />





let $e9051af08b8b1500$export$bca14c5b3b88a9c9 = {
    global: ($parcel$interopDefault($33116a1ac1f127f3$exports)),
    light: (0, ($parcel$interopDefault($2179ca4439c0728e$exports))),
    dark: (0, ($parcel$interopDefault($5fc600550714d659$exports))),
    medium: ($parcel$interopDefault($7ada9fcc793cf5cf$exports)),
    large: ($parcel$interopDefault($ed6c61a7b1f80a21$exports))
};
