// @react-spectrum/table/dist/import.mjs
import {TableView as $9f67189e3f884740$export$b3c27e869d856b7} from "./TableViewWrapper.mjs";
import {Column as $gyhAV$Column, TableHeader as $e668f3d0f378a712$re_export$TableHeader, TableBody as $e668f3d0f378a712$re_export$TableBody, Section as $e668f3d0f378a712$re_export$Section, Row as $e668f3d0f378a712$re_export$Row, Cell as $e668f3d0f378a712$re_export$Cell} from "@react-stately/table";

/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />


// Override TS for Column to support spectrum specific props.
const $e668f3d0f378a712$export$816b5d811295e6bc = (0, $gyhAV$Column);


export {$e668f3d0f378a712$export$816b5d811295e6bc as Column, $9f67189e3f884740$export$b3c27e869d856b7 as TableView, $e668f3d0f378a712$re_export$TableHeader as TableHeader, $e668f3d0f378a712$re_export$TableBody as TableBody, $e668f3d0f378a712$re_export$Section as Section, $e668f3d0f378a712$re_export$Row as Row, $e668f3d0f378a712$re_export$Cell as Cell};
